app/helpers/review_mapping_helper.rb

```diff
@@ -0,0 +1,123 @@
+module ReviewMappingHelper
+  def create_report_table_header(headers = {})
+    table_header = "<div class = 'reviewreport'>\
+                    <table width='100% cellspacing='0' cellpadding='2' border='0'>\
+                    <tr bgcolor='#CCCCCC'>"
+    headers.each do |header, percentage|
+      table_header += "<th width = #{percentage}>\
+                      #{header.humanize}\
+                      </th>"
+    end
+    table_header += "</tr>"
+    table_header.html_safe
+  end
+
+  #
+  # for review report
+  #
+  def get_data_for_review_report(reviewed_object_id, reviewer_id, type, line_num)
+    rspan = 0
+    line_num += 1
+    bgcolor = line_num.even? ? "#ffffff" : "#DDDDBB"
+    (1..@assignment.num_review_rounds).each {|round| instance_variable_set("@review_in_round_" + round.to_s, 0) }
+
+    response_maps = ResponseMap.where(["reviewed_object_id = ? AND reviewer_id = ? AND type = ?", reviewed_object_id, reviewer_id, type])
+    response_maps.each do |ri|
+      rspan += 1 if Team.exists?(id: ri.reviewee_id)
+      responses = ri.response
+      (1..@assignment.num_review_rounds).each do |round|
+        instance_variable_set("@review_in_round_" + round.to_s, instance_variable_get("@review_in_round_" + round.to_s) + 1) if responses.exists?(round: round)
+      end
+    end
+    [response_maps, bgcolor, rspan, line_num]
+  end
+
+  def get_team_reviewed_link_name(max_team_size, response, reviewee_id)
+    team_reviewed_link_name = if max_team_size == 1
+                                TeamsUser.where(team_id: reviewee_id).first.user.fullname
+                              else
+                                Team.find(reviewee_id).name
+                              end
+    team_reviewed_link_name = "(" + team_reviewed_link_name + ")" if !response.empty? and !response.last.is_submitted?
+    team_reviewed_link_name
+  end
+
+  def get_current_round_for_review_report(reviewer_id)
+    user_id = Participant.find(reviewer_id).user.id
+    topic_id = SignedUpTeam.topic_id(@assignment.id, user_id)
+    current_round = @assignment.number_of_current_round(topic_id)
+    current_round = @assignment.num_review_rounds if @assignment.get_current_stage(topic_id) == "Finished" || @assignment.get_current_stage(topic_id) == "metareview"
+  end
+
+  # varying rubric by round
+  def get_each_round_score_awarded_for_review_report(reviewer_id, team_id)
```

Comment:
Assignment Branch Condition size for get_each_round_score_awarded_for_review_report is too high. [25.32/15]